engine/evaluator.go

```diff
@@ -103,7 +103,13 @@ func CheckFourOfAKind(hand []Card) (bool, Tiebreakers) {
 	if len(quads) != 1 {
 		return false, nil
 	}
-	return true, Tiebreakers{quads[0]}
+	kicker := 0
+	for _, rank := range handRanks {
+		if rank != quads[0] && rank > kicker {
+			kicker = rank
+		}
+	}
+	return true, Tiebreakers{quads[0], kicker}
```

Comment:
Great catch! While this is correct, could we do this in the same way that we do it in `CheckTwoPair` for consistency? Something like this:

```go

	kickerRanks := getCardRankIdSlice(hand)
	kickerRanks = filterInt(kickerRanks, func(i int) bool {
		return i != quads[0]
	})
	descendingSort(kickerRanks)
	return true, Tiebreakers{quads[0], kickerRanks[0]}
```

Reply:
Sure, but isn't `kickerRanks := getCardRankIdSlice(hand)` redundant? You could just use the handRanks var if I'm not mistaken.

Reply:
Yes, I noticed that. However, `getFrequencies` modifies `kickerRanks` in-place (with `Sort`), and I want to avoid reusing variables that could be modified in child functions.

I think a better solution would be to make sure that `getFrequencies` doesn't modify its inputs and then we should be able to reuse `handRanks` here and in `CheckThreeOfAKind`, `CheckTwoPair`, and `CheckPair`. 

Here is how `getFrequencies` would get refactored:
```go
func getFrequencies(list []int, k, j int) []int {
	sortList := list
	sort.Sort(sort.Reverse(sort.IntSlice(sortList)))
	var matches []int
	count := 0
	for i, current := range sortList {
		count++
		if i == len(sortList)-1 || sortList[i+1] != current {
			if count >= k && count <= j {
				matches = append(matches, current)
			}
			count = 0
		}
	}
	return matches
}
```

What do you think?

Reply:
To be honest with you, I haven't read much of your code. I just helped a friend of mine that's using it fix the issue with four of a kind. You definitely know alot more about this project than me and if you think it's the best way forward then go for it.